presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainOrcPredicate.java

```diff
@@ -75,9 +96,149 @@ public boolean matches(long numberOfRows, Map<Integer, ColumnStatistics> statist
+            if (domain.isAll()) {
+                // all values are okay, go to next one as we don't have to check the bf
+                continue;
+            }
+
+            // extract values from domain
+            Collection<Object> predicateValues = predicateValuesFromDomain(domain);
+            if (predicateValues == null || predicateValues.isEmpty()) {
+                // no values checked, treat as failure: read
+                return true;
+            }
+
+            // run values against the bloomfilters: none can match in order to skip a read
+            for (Object predicateValue : predicateValues) {
+                for (RowGroupBloomfilter rowGroupBloomfilter : bloomfilters) {
+                    BloomFilter bloomfilter = rowGroupBloomfilter.getBloomfilter();
+
+                    boolean matched = checkInBloomFilter(bloomfilter, predicateValue);
+                    numberOfBloomFiltersChecked++;
+                    if (matched) {
+                        // bloom filter is matched here return true so we select this stripe as it likely contains data which we need to read
+                        return true;
+                    }
+                }
+            }
+        }
+
+        // did we read any bloom filter?
+        if (numberOfBloomFiltersChecked < 1) {
+            // no bloom filters read, seems like a query that was run only on partitioned columns
+            return true;
+        }
+
+        // none of the bloomfilters caused a "hit" meaning we should not read
+        log.debug("Not reading, didn't match any of the bloomfilters (" + numberOfBloomFiltersChecked + "x), data is not here");
+        return false;
+    }
+
+    @VisibleForTesting
+    public static Collection<Object> predicateValuesFromDomain(Domain domain)
+    {
+        ValueSet values = domain.getValues();
+        if (values instanceof EquatableValueSet) {
+            EquatableValueSet eqValues = (EquatableValueSet) values;
+            if (eqValues.isWhiteList()) {
+                // we can only work with values we know, not excluded blacklists because other rows might contain the data we need
+                return ImmutableSet.of(values.getDiscreteValues().getValues());
+            }
+        }
+        else if (values instanceof SortedRangeSet) {
+            SortedRangeSet sortedRangeSet = (SortedRangeSet) values;
+            // sorted range set is used for integer comparison (e.g. id = 123 ) where min and max is the same value
+            if (sortedRangeSet.isSingleValue()) {
+                return ImmutableSet.of(sortedRangeSet.getSingleValue());
+            }
+        }
+        return ImmutableSet.of();
+    }
+
+    // checks whether a value part of the effective predicate is likely to be part of this bloomfilter
+    @VisibleForTesting
+    public static boolean checkInBloomFilter(BloomFilter bloomfilter, Object predicateValue)
+    {
+        if (predicateValue instanceof Long) {
```

Comment:
Interesting, haven't thought of that. Code was inspired on https://github.com/apache/orc/blob/master/java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java#L556 because that's how the regular Hive works, and I thought to keep it so similar possible to prevent issues. What would you propose?